src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -90,6 +90,11 @@ public function __construct(ParameterBagInterface $parameterBag = null)
         $this->parameterBag = $parameterBag ?: new ParameterBag();
     }
 
+    public function __clone()
```

Comment:
I suggest using a private (final) ``__clone`` method. This way, PHP itself will figure out that the class cannot be cloned (``ReflectionClass::isCloneable`` will return ``false``)